src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,105 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] elementData = (T[]) new Object[DEFAULT_CAPACITY];
+    private int size;
 
     @Override
     public void add(T value) {
-
+        if (size + 1 >= elementData.length) {
+            T[] newArray = (T[]) new Object[changeLength()];
+            System.arraycopy(elementData, 0, newArray, 0, size);
+            elementData = newArray;
+        }
+        elementData[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("Wrong index");
+        }
+        if (index == size) {
+            add(value);
+            return;
+        }
+        if (size == elementData.length) {
+            grow(value,index);
+        }
+        System.arraycopy(elementData, index, elementData, index + 1, size - index);
+        elementData[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index < 0 || index > size - 1) {
+            throw new ArrayListIndexOutOfBoundsException("Wrong index");
+        }
+        return elementData[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        if (index > size - 1 || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Wrong index");
+        }
+        elementData[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        Object removeElement;
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("Wrong index");
+        }
+        removeElement = elementData[index];
+        int needNumberCopy = size - index - 1;
+        System.arraycopy(elementData, index + 1, elementData, index, needNumberCopy);
+        size--;
+        return (T) removeElement;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        for (int i = 0; i < size; i++) {
+            if (elementData[i] == element || elementData[i] != null
+                    && element.equals(elementData[i])) {
+                return remove(i);
+            }
+        }
+        throw new NoSuchElementException("No such element");
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private int changeLength() {
+        return elementData.length
+                + (elementData.length >> 1);
+    }
+
+    private void grow(T value, int index) {
```

Comment:
Змінюй лише розмір масиву, не змінюй значення елементів масиву в цьому методі
```suggestion
    private void grow() {
```